### PR TITLE
[fix]: subflow execution with .#1

### PR DESCRIPTION
**Pull request by yashsinghcodes**
fix for Shuffle/Shuffle#1463

### CODE REVIEW

**Comment by yashsinghcodes** [comment]
also fixes the Shuffle/Shuffle#1492 for shuffle-shared as it was required to actually test my file.

**Comment by yashsinghcodes** [comment]
@frikky please have a look :)

File: shared_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    handlers := []struct {
+        arg         string
+        expected    bool
+    }{
+        {"$exec.#1-2", true},
+        {"$exec.#.value.#1", true},
+        {"$exec.#1", false},
+        {"$exec", false},
+        {"$exec.#1.value.#2", false},
+        {"$start_node.#", true},
```
Comment (frikky) [inline shared_test.go:17]: Remember some newline tests ;)

File: shared.go [hunk 2]
```diff
@@ -159,6 +159,15 @@ func randStr(strSize int, randType string) string {
 	return string(bytes)
 }
 
+func isLoop(arg string) bool {
+    if strings.Contains(arg,"$") && strings.Contains(arg, ".#") {
+        pattern := `(^|\.)(#(\d+-\d+)?($|\.))`
```
Comment (frikky) [inline shared.go:164]: Can we add another if inside the other if that checks for string contains  ".#." or string endswith(".#") and just return true if it is?

I want to use the regex as little as possible :)

**Pull request merged**